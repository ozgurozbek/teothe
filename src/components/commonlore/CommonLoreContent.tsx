"use client";
import { Button, Card, Divider } from "antd";
import GetCrumbs from "@/components/NavigationCrumb";
import SimpleContent from "@/components/SimpleCon";
import { DoubleRightOutlined } from "@ant-design/icons";
/**
 * Uses SimpleContent to generate a page full of text describing the common lore of teothe.
 * @tutorial
 * @see SimpleContent
 * @returns Card(antd) wrapped by section
 */
export default function CommonLoreContent() {
  return (
    <section>
      <GetCrumbs path={"Teothe,Common Lore"} />
      <Card bordered={false} className="w-full">
        <SimpleContent
          contentProps={{
            title: "Intro",
            text: [
              'Long before their battles, gods were once "All," harnessing the energy and magic of reality. However, their egos led to conflicts and the need to determine superiority. Lesser creatures desired an end to the fighting, but gods, being immortal and unforgiving, couldn\'t accept another "Perfect Being\'s" actions.',
              "Eventually, prayers were answered, and the gods ceased fighting, but this created a worse situation. They stopped communicating and individually created humans from Mceeju, who, despite their differences, served the gods. The gods made peace with their divergences and established various species within the Astral Sea.",
              "Humans, lacking intelligence and divine guidance, possessed magic but often perished due to imperfections. Their deaths released magic into the universe. The gods accepted their punishment and judged themselves, losing their perfect nature and becoming distinct from one another.",
              "The gods continued to create humans, who exhibited predictable yet mysterious traits. They showed mercy and guided some humans towards greatness, rewarding them with demigod status. However, such accomplishments became scarce over time.",
              "Despite sharing the same creators, species clashed or multiplied uncontrollably, losing control of the Weave and facing an unstoppable force: Mceeju.",
              "Mceeju divided the Astral Sea, creating the Material Planes and the Inner Worlds while isolating the Elemental Chaos with a dimensional barrier. Species were divided, and even the gods' domains were split. The Overworld, Underworld, and Outer Worlds emerged.",
              "With newfound freedom, the gods led by Mceeju's order. Some stayed united, some pursued separate paths, and others descended to the Material Planes.",
            ],
          }}
        />
        <Divider />
        <SimpleContent
          contentProps={{
            image: {
              src: "./wheel.png",
              alt: "Wheel Cosmology",
            },
            title: "Dimensional Cosmology",
            text: [
              "The Wheel Cosmology in Teothe is a conceptual framework that organizes the multiverse into a series of interconnected planes and realms. It envisions existence as a great wheel with various planes revolving around a central hub, the Marble. The cosmology encompasses the Far Realms, Outer Planes representing different alignments, Inner Planes embodying elemental forces, the Astral Sea connecting all planes, and additional planes like the Feywild and Shadowfell that mirror and intersect with the Material Plane. This cosmological model provides a structured and interconnected backdrop for adventures and campaigns, offering a vast array of realms and dimensions for exploration and storytelling.",
              "Material Plane: The planes of existence where most D&D adventures take place. It is a diverse and varied world, consisting of continents, seas, and skies. It is the realm inhabited by mortals, and it serves as a central hub connecting with other planes. Teothe is one of many material planes, inhabited by a diverse selection of species across all continents.",
              "Feywild: Also known as the Plane of Faerie, it is a vibrant and magical reflection of the Material Plane. It is home to fey creatures, such as fairies, elves, and other enchanted beings. Time and nature behave differently in the Feywild.",
              "Shadowfell: A dark and gloomy plane that mirrors the Material Plane, associated with darkness, decay, and negative emotions. It is a realm of shadows and is inhabited by undead creatures and other dark entities. Time and nature behave differently in the Shadow plane.",
              "Ethereal Plane: A misty and insubstantial realm that coexists with the Material Plane. It is often accessed through magical means and is used for travel and interaction with ethereal beings and objects.",
              "Elemental Planes: In addition to the Inner Planes, there are also Para-Elemental Planes and Quasi-Elemental Planes. The Para-Elemental Planes lie between two adjacent Inner Planes and contain a blend of elemental energies. The Quasi-Elemental Planes exist between the Elemental Planes and combine elemental forces with other phenomena like ash, steam, ice, and more.",
              "Inner Planes: Elemental planes that embody the raw forces of nature. They include the Elemental Plane of Fire, Elemental Plane of Water, Elemental Plane of Air, and Elemental Plane of Earth. Each plane is abundant in its corresponding elemental energy.",
              "Outer Planes: A collection of planes representing different moral and ethical alignments. These planes include the Overseer realms. They are home to powerful deities and beings embodying these alignments.",
              "Astral Sea: The plane that connects all the other planes, serving as a vast void of silvery mist. It is traversed by the Astral Projection and acts as a conduit between different realms. It is also home to astral creatures.",
              "Far Realms: A plane of existence that lies beyond the known multiverse, characterized by eldritch and incomprehensible entities. It is a realm of madness and alien horrors. Even sympathizing with such concepts is forbidden in most material planes including Teothe.",
              "These summaries provide a brief overview of each component within the Dimensional Cosmology of Teothe, offering a glimpse into the diverse and interconnected planes that players and characters may explore during their adventures.",
            ],
          }}
        />
        <Button href="/planes">
          <span>
            See Planes <DoubleRightOutlined />
          </span>
        </Button>
        <Divider />
        <SimpleContent
          contentProps={{
            title: "Gate Keepers",
            text: [
              "All take their power from Mceeju so all are close to gods in power. In Teothe, Gate Keepers are a topic of interest for many sages and very little about them is known.",
              "Images of Raajod are the gate-keepers of Overworld Vocals that are in Outer Realms for many marbles.",
              "Images of Raxy are the gate-keepers of Underworld Vocals that are in Outer Realms for many marbles.",
              "Onposufm or Wefhiza maintains order in the Astral Sea, but is believed to be the guardian of accidents.",
            ],
          }}
        />
        <Divider />
        <SimpleContent
          contentProps={{
            title: "Ancients",
            text: [
              "Ancients all take their power from Mceeju so all are close to gods in power. People of Teothe, the ones who know about ancients, believe all of them to be perished over time. ",
              "An old civilization that has evolved and found great meaning in both magic and immortality. They made immortality possible and destroyed every meaning they've found within it. With their last magic, they left a warning for future, unfortunately too difficult to grasp, lacking their understanding of magic, life and time. They managed to process light, and believed to achieve a hive-mind society. Nobody knows how they disappeared.",
              "Their offsprings from humans tend to get a disoriented body with partial ancient bodyparts. They live for about a millenia years without aging beyond their thirties, body-wise. They are sterile. Even though some were legends, no half-ancient have been spotted in Teothe in centuries.",
              "The last Ancient Relic dwindled out of existance, causing the Blue Era in Teothe. Steram Wayogo, known as the last half-ancient embarked on a journey across countless marbles sealing Ancient Gates, that once provided a unique magical flow to the Material Planes.",
            ],
          }}
        />
        <Divider />
        <SimpleContent
          contentProps={{
            title: "Domains",
            text: [
              "Deities of Teothe are divided into 8 domains. These domains are Nature, Tempest, Trickery, Death, Knowledge, War, Life and Light. Most Deities exist within one to two domains.",
              "Pantheon deities use fundamental domains. Gameplay-wise, other domains like Endless domains (destruction, destiny, despair, desire, delirium, dream), Broad domains (decay, drama, fabrication, fate, time, distance, potential) or ones introduced in XGE (forge, grave), or UA (order) are welcome.",
            ],
          }}
        />
        <Button href="/deities">
          <span>
            See Deities <DoubleRightOutlined />
          </span>
        </Button>
      </Card>
    </section>
  );
}
