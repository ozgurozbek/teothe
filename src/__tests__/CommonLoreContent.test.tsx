import React from 'react';
import { render } from '@testing-library/react';
import CommonLoreContent from '../components/commonlore/CommonLoreContent';

describe('CommonLoreContent', () => {
  test('renders without crashing', () => {
    render(<CommonLoreContent />);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<CommonLoreContent />);
    expect(asFragment()).toMatchSnapshot();
  });

});
