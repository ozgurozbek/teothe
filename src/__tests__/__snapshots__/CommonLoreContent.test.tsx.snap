// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonLoreContent matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <span
      class="text-slate-400"
    >
       / Teothe
    </span>
    <span>
       / Common Lore
    </span>
    <div
      class="ant-card w-full css-dev-only-do-not-override-1xg9z9n"
    >
      <div
        class="ant-card-body"
      >
        <section>
          <h1
            class="ant-typography css-dev-only-do-not-override-1xg9z9n"
            data-testid="simplecon-title"
            id="intro"
          >
            Intro
          </h1>
          <div
            data-testid="simplecon-paragraph"
          >
            <p
              class="mb-2"
            >
              Long before their battles, gods were once "All," harnessing the energy and magic of reality. However, their egos led to conflicts and the need to determine superiority. Lesser creatures desired an end to the fighting, but gods, being immortal and unforgiving, couldn't accept another "Perfect Being's" actions.
            </p>
            <p
              class="mb-2"
            >
              Eventually, prayers were answered, and the gods ceased fighting, but this created a worse situation. They stopped communicating and individually created humans from Mceeju, who, despite their differences, served the gods. The gods made peace with their divergences and established various species within the Astral Sea.
            </p>
            <p
              class="mb-2"
            >
              Humans, lacking intelligence and divine guidance, possessed magic but often perished due to imperfections. Their deaths released magic into the universe. The gods accepted their punishment and judged themselves, losing their perfect nature and becoming distinct from one another.
            </p>
            <p
              class="mb-2"
            >
              The gods continued to create humans, who exhibited predictable yet mysterious traits. They showed mercy and guided some humans towards greatness, rewarding them with demigod status. However, such accomplishments became scarce over time.
            </p>
            <p
              class="mb-2"
            >
              Despite sharing the same creators, species clashed or multiplied uncontrollably, losing control of the Weave and facing an unstoppable force: Mceeju.
            </p>
            <p
              class="mb-2"
            >
              Mceeju divided the Astral Sea, creating the Material Planes and the Inner Worlds while isolating the Elemental Chaos with a dimensional barrier. Species were divided, and even the gods' domains were split. The Overworld, Underworld, and Outer Worlds emerged.
            </p>
            <p
              class="mb-2"
            >
              With newfound freedom, the gods led by Mceeju's order. Some stayed united, some pursued separate paths, and others descended to the Material Planes.
            </p>
          </div>
        </section>
        <div
          class="ant-divider css-dev-only-do-not-override-1xg9z9n ant-divider-horizontal"
          role="separator"
        />
        <section>
          <h1
            class="ant-typography css-dev-only-do-not-override-1xg9z9n"
            data-testid="simplecon-title"
            id="dimensional-cosmology"
          >
            Dimensional Cosmology
          </h1>
          <div
            class="ant-image w-full css-dev-only-do-not-override-1xg9z9n"
          >
            <img
              alt="Wheel Cosmology"
              class="ant-image-img rounded-xl w-full css-dev-only-do-not-override-1xg9z9n"
              src="./wheel.png"
            />
            <div
              class="ant-image-mask"
            >
              <div
                class="ant-image-mask-info"
              >
                <span
                  aria-label="eye"
                  class="anticon anticon-eye"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="eye"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                    />
                  </svg>
                </span>
                Preview
              </div>
            </div>
          </div>
          <div
            data-testid="simplecon-paragraph"
          >
            <p
              class="mb-2"
            >
              The Wheel Cosmology in Teothe is a conceptual framework that organizes the multiverse into a series of interconnected planes and realms. It envisions existence as a great wheel with various planes revolving around a central hub, the Marble. The cosmology encompasses the Far Realms, Outer Planes representing different alignments, Inner Planes embodying elemental forces, the Astral Sea connecting all planes, and additional planes like the Feywild and Shadowfell that mirror and intersect with the Material Plane. This cosmological model provides a structured and interconnected backdrop for adventures and campaigns, offering a vast array of realms and dimensions for exploration and storytelling.
            </p>
            <p
              class="mb-2"
            >
              Material Plane: The planes of existence where most D&D adventures take place. It is a diverse and varied world, consisting of continents, seas, and skies. It is the realm inhabited by mortals, and it serves as a central hub connecting with other planes. Teothe is one of many material planes, inhabited by a diverse selection of species across all continents.
            </p>
            <p
              class="mb-2"
            >
              Feywild: Also known as the Plane of Faerie, it is a vibrant and magical reflection of the Material Plane. It is home to fey creatures, such as fairies, elves, and other enchanted beings. Time and nature behave differently in the Feywild.
            </p>
            <p
              class="mb-2"
            >
              Shadowfell: A dark and gloomy plane that mirrors the Material Plane, associated with darkness, decay, and negative emotions. It is a realm of shadows and is inhabited by undead creatures and other dark entities. Time and nature behave differently in the Shadow plane.
            </p>
            <p
              class="mb-2"
            >
              Ethereal Plane: A misty and insubstantial realm that coexists with the Material Plane. It is often accessed through magical means and is used for travel and interaction with ethereal beings and objects.
            </p>
            <p
              class="mb-2"
            >
              Elemental Planes: In addition to the Inner Planes, there are also Para-Elemental Planes and Quasi-Elemental Planes. The Para-Elemental Planes lie between two adjacent Inner Planes and contain a blend of elemental energies. The Quasi-Elemental Planes exist between the Elemental Planes and combine elemental forces with other phenomena like ash, steam, ice, and more.
            </p>
            <p
              class="mb-2"
            >
              Inner Planes: Elemental planes that embody the raw forces of nature. They include the Elemental Plane of Fire, Elemental Plane of Water, Elemental Plane of Air, and Elemental Plane of Earth. Each plane is abundant in its corresponding elemental energy.
            </p>
            <p
              class="mb-2"
            >
              Outer Planes: A collection of planes representing different moral and ethical alignments. These planes include the Overseer realms. They are home to powerful deities and beings embodying these alignments.
            </p>
            <p
              class="mb-2"
            >
              Astral Sea: The plane that connects all the other planes, serving as a vast void of silvery mist. It is traversed by the Astral Projection and acts as a conduit between different realms. It is also home to astral creatures.
            </p>
            <p
              class="mb-2"
            >
              Far Realms: A plane of existence that lies beyond the known multiverse, characterized by eldritch and incomprehensible entities. It is a realm of madness and alien horrors. Even sympathizing with such concepts is forbidden in most material planes including Teothe.
            </p>
            <p
              class="mb-2"
            >
              These summaries provide a brief overview of each component within the Dimensional Cosmology of Teothe, offering a glimpse into the diverse and interconnected planes that players and characters may explore during their adventures.
            </p>
          </div>
        </section>
        <a
          class="ant-btn css-dev-only-do-not-override-1xg9z9n ant-btn-default"
          href="/planes"
          tabindex="0"
        >
          <span>
            See Planes 
            <span
              aria-label="double-right"
              class="anticon anticon-double-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="double-right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                />
              </svg>
            </span>
          </span>
        </a>
        <div
          class="ant-divider css-dev-only-do-not-override-1xg9z9n ant-divider-horizontal"
          role="separator"
        />
        <section>
          <h1
            class="ant-typography css-dev-only-do-not-override-1xg9z9n"
            data-testid="simplecon-title"
            id="gate-keepers"
          >
            Gate Keepers
          </h1>
          <div
            data-testid="simplecon-paragraph"
          >
            <p
              class="mb-2"
            >
              All take their power from Mceeju so all are close to gods in power. In Teothe, Gate Keepers are a topic of interest for many sages and very little about them is known.
            </p>
            <p
              class="mb-2"
            >
              Images of Raajod are the gate-keepers of Overworld Vocals that are in Outer Realms for many marbles.
            </p>
            <p
              class="mb-2"
            >
              Images of Raxy are the gate-keepers of Underworld Vocals that are in Outer Realms for many marbles.
            </p>
            <p
              class="mb-2"
            >
              Onposufm or Wefhiza maintains order in the Astral Sea, but is believed to be the guardian of accidents.
            </p>
          </div>
        </section>
        <div
          class="ant-divider css-dev-only-do-not-override-1xg9z9n ant-divider-horizontal"
          role="separator"
        />
        <section>
          <h1
            class="ant-typography css-dev-only-do-not-override-1xg9z9n"
            data-testid="simplecon-title"
            id="ancients"
          >
            Ancients
          </h1>
          <div
            data-testid="simplecon-paragraph"
          >
            <p
              class="mb-2"
            >
              Ancients all take their power from Mceeju so all are close to gods in power. People of Teothe, the ones who know about ancients, believe all of them to be perished over time. 
            </p>
            <p
              class="mb-2"
            >
              An old civilization that has evolved and found great meaning in both magic and immortality. They made immortality possible and destroyed every meaning they've found within it. With their last magic, they left a warning for future, unfortunately too difficult to grasp, lacking their understanding of magic, life and time. They managed to process light, and believed to achieve a hive-mind society. Nobody knows how they disappeared.
            </p>
            <p
              class="mb-2"
            >
              Their offsprings from humans tend to get a disoriented body with partial ancient bodyparts. They live for about a millenia years without aging beyond their thirties, body-wise. They are sterile. Even though some were legends, no half-ancient have been spotted in Teothe in centuries.
            </p>
            <p
              class="mb-2"
            >
              The last Ancient Relic dwindled out of existance, causing the Blue Era in Teothe. Steram Wayogo, known as the last half-ancient embarked on a journey across countless marbles sealing Ancient Gates, that once provided a unique magical flow to the Material Planes.
            </p>
          </div>
        </section>
        <div
          class="ant-divider css-dev-only-do-not-override-1xg9z9n ant-divider-horizontal"
          role="separator"
        />
        <section>
          <h1
            class="ant-typography css-dev-only-do-not-override-1xg9z9n"
            data-testid="simplecon-title"
            id="domains"
          >
            Domains
          </h1>
          <div
            data-testid="simplecon-paragraph"
          >
            <p
              class="mb-2"
            >
              Deities of Teothe are divided into 8 domains. These domains are Nature, Tempest, Trickery, Death, Knowledge, War, Life and Light. Most Deities exist within one to two domains.
            </p>
            <p
              class="mb-2"
            >
              Pantheon deities use fundamental domains. Gameplay-wise, other domains like Endless domains (destruction, destiny, despair, desire, delirium, dream), Broad domains (decay, drama, fabrication, fate, time, distance, potential) or ones introduced in XGE (forge, grave), or UA (order) are welcome.
            </p>
          </div>
        </section>
        <a
          class="ant-btn css-dev-only-do-not-override-1xg9z9n ant-btn-default"
          href="/deities"
          tabindex="0"
        >
          <span>
            See Deities 
            <span
              aria-label="double-right"
              class="anticon anticon-double-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="double-right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                />
              </svg>
            </span>
          </span>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
